src/models/department.ts

```diff
@@ -0,0 +1,19 @@
+import Position from "./position";
+import Employee from "./employee";
+import { Prop, PropType, Ref } from "@typegoose/typegoose";
+
+class Department {
+    @Prop({ required: true, type: () => String }) //? Should maybe be unique?
+    public name!: string;
+
+    @Prop({ required: true, ref: () => Position, type: () => Position })
+    public departmentHead!: Ref<Position>;
+
+    @Prop({ required: true, ref: () => Position, type: () => Position })
+    public leadershipTeamMember!: Ref<Position>;
+
+    @Prop({ required: true, ref: () => Employee, type: () => [Employee] }, PropType.ARRAY)
+    public employees: Ref<Employee>[] = [];
```

Comment:
I'm not sure that this array is necessary - I think it was just in our TypeORM type because of their stuff

Reply:
Yeah I don't think we need it necessarily but tbh I think it might be helpful later on to have just a list of employees under a department?

Reply:
That makes sense, we can keep it :)